packages/plugins/visitor-plugin-common/src/naming.ts

```diff
@@ -0,0 +1,95 @@
+import { ASTNode } from 'graphql';
+import { resolveExternalModuleAndFn } from 'graphql-codegen-plugin-helpers';
+import { NamingConventionMap, NamingConvention, ConvertFn } from './types';
+import { toPascalCase } from './utils';
+
+function getKind(node: ASTNode | string): keyof NamingConventionMap {
+  if (typeof node === 'string') {
+    return 'typeNames';
+  }
+
+  if (['EnumValueDefinition', 'EnumValue'].includes(node.kind)) {
+    return 'enumValues';
+  }
+
+  return 'typeNames';
+}
+
+function getName(node: ASTNode | string): string | undefined {
+  if (typeof node === 'string') {
+    return node;
+  }
+
+  switch (node.kind) {
+    case 'OperationDefinition':
+    case 'Variable':
+    case 'Argument':
+    case 'FragmentSpread':
+    case 'FragmentDefinition':
+    case 'ObjectField':
+    case 'Directive':
+    case 'NamedType':
+    case 'ScalarTypeDefinition':
+    case 'ObjectTypeDefinition':
+    case 'FieldDefinition':
+    case 'InputValueDefinition':
+    case 'InterfaceTypeDefinition':
+    case 'UnionTypeDefinition':
+    case 'EnumTypeDefinition':
+    case 'EnumValueDefinition':
+    case 'InputObjectTypeDefinition':
+    case 'DirectiveDefinition': {
+      return getName(node.name);
+    }
+    case 'Name': {
+      return node.value;
+    }
+    case 'Field': {
+      return getName(node.alias || node.name);
+    }
+    case 'VariableDefinition': {
+      return getName(node.variable);
+    }
+  }
+
+  return undefined;
+}
+
+export function convertFactory(config: { namingConvention?: NamingConvention }): ConvertFn {
+  function resolveConventionName(type: keyof NamingConventionMap): (str: string) => string {
+    if (!config.namingConvention) {
```

Comment:
Allow it to be `function` as well, because we can't run `resolveExternalModuleAndFn` when we are running in a browser.